fix(sso): Update and simplify SSO docs (#1487)
getsentry/sentry-docs: src/collections/_documentation/server/sso.md

-   Via a middleware which handles an upstream proxy dictating the authenticated user
-   Via a third-party service which implements an authentication pipeline

> This documentation describes the latter, which would cover things like Google Apps and GitHub.

Reviewer: ```suggestion
This documentation describes the latter, which will cover topics like Google Apps and GitHub.
```

> 

Reviewer 2: @BYK curious why we are removing the example of Saml2 and rippling?

Author: This is not an example, it is just showing a feature flag that is obsolete as it is enabled by default for a while now.

Author: @EvanPurkhiser it is obsolete for on-premise as it is enabled by default there.

## Installing a Provider

### Sentry >= 10

- **GitHub** is bundled with Sentry. Follow the [GitHub integration guide]({%- link _documentation/server/integrations/github/index.md -%}) and then install your GitHub app to your organization.

### Sentry >= 9.1

- **Google Auth** is bundled with Sentry. Set the following values in your `config.yml` file:

```yaml
auth-google.client-id: '<client id>'
auth-google.client-secret: '<client secret>'
```

## Custom Providers

> At this time the API is considered unstable and subject to change. Things likely won’t change a lot, but there’s a few areas that need some clean up.

Reviewer: ```suggestion
At this time, the API is considered unstable and subject to change. There won't be many changes, but a few areas need some cleanup.
```
